x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -101,10 +96,23 @@ export async function getTags({
   savedObjectsClient: soClient,
   caseService,
   logger,
+  authorization: auth,
+  auditLogger,
 }: CasesClientArgs): Promise<string[]> {
   try {
+    let authFindHelpers: AuthorizationFilter;
+    try {
+      authFindHelpers = await auth.getFindAuthorizationFilter(CASE_SAVED_OBJECT);
+    } catch (error) {
+      auditLogger?.log(createAuditMsg({ operation: Operations.getTags, error }));
+      throw error;
+    }
+
+    const { filter: authorizationFilter } = authFindHelpers;
+
     return await caseService.getTags({
       soClient,
+      filter: authorizationFilter,
```

Comment:
hmm I hadn't thought about this scenario. Let's say user Jon has access to both `securitySolution` and `observability`. Then Jon accesses the cases list view through security solution where a request to `getTags` is made. Do we want to show _all_ tags that Jon has privileges to view? Or just the tags that make sense within the security solution app?

Basically I'm wondering if we should expose the `filter` parameter in the REST API and the UI will need to pass a value down, otherwise we'll return everything they're authorized to view?

Reply:
You are right! I missed it 😄